core/tee/tee_svc_storage.c

```diff
@@ -626,6 +627,9 @@ TEE_Result syscall_storage_obj_create(unsigned long storage_id, void *object_id,
 		goto err;
 	}
 
+	/* remove the file if it exists*/
+	if (!filedoesnotexist)
+		tee_file_ops.unlink(file);
```

Comment:
@pascal-brand-st-dev, it's better to have this fix than not even if it's not atomic, given that we plan to address that later.
Please add a comment as you propose to highlight the problem.